### ISSUE TITLE
Initializing enum members with another string enum causes incorrect/inconsistent types

### ISSUE THREAD

**Issue by devrelm** [microsoft/TypeScript#26522]
**TypeScript Version:**  2.4.1 through 3.1.0-dev.20180817

**Search Terms:**
* string enum values act like numbers
* inconsistent string enum types
* string enum of string enum

**Code**

```ts
enum Foo {
    Foo1 = 'hello',
    Foo2 = 'goodbye',
}

enum Bar {
    Bar1 = Foo.Foo1,
    Bar2 = Foo.Foo2,
}

const myStringFunc = (baz: string) => console.log(baz);

const myFooFunc = (baz: Foo) => myStringFunc(baz);

const myBarFunc = (baz: Bar) => myStringFunc(baz);
                                          // ~~~ Argument of type 'Bar' is not
                                          // assignable to parameter of type 'string'.

// This compiles fine! TypeScript thinks baz is a number!
const myOddFunc = (baz: Bar) => baz.toExponential();
```

**Expected behavior:**
String enums like `Bar` that are initialized with other string enum values should be properly typed as a string enum.

**Actual behavior:**
`Bar` is inconsistently typed as a numeric enum, as evidenced by `baz.toExponential()` not causing a compiler error.

To make it more fun, the members of `Bar` even autocomplete and hint as the correct string literals:
![image](https://user-images.githubusercontent.com/2008333/44288738-4b9cb280-a23f-11e8-91fc-144cfd6dd1ed.png)
![image](https://user-images.githubusercontent.com/2008333/44288843-be0d9280-a23f-11e8-8256-7fa1c6458784.png)

But the members of `Bar` themselves autocomplete members of `Number.prototype`:
![image](https://user-images.githubusercontent.com/2008333/44289400-062db480-a242-11e8-95d3-c529f89d4275.png)

**Playground Link:**
[here](https://agentcooper.github.io/typescript-play/#code/KYOwrgtgBAYg9nKBvAUFdsEEYoF4oDkAFsADalwEA0aG8cATHoQOYIAmARgJ7DUoBfFClCQoAIQCGAJ2S10U6Tnz0AdPSw0MEmUxUJ1CBjSEoAxnBABnAC5QI3AMo3pASxAsYYEGeYAKTkkALwAuKFs3DwBKPAA+KAtrOFJgVQoWAOCogG5hRNt7bnovH39A0Mw4GNx4h2dIz28zTKCcvMsCh0US33wWsMVq2qcXd0afFrbtaZnZufmMAHpFqAA-dagAQWkWSFA7OAAzKBtuAAdgQkUCKFcrKBA4G3kF17eZ5ahJKytXFhBJJwUidEGcZJIIMAbMBZEcTudLgQImMCKphJ8ACpEO4JOAQM6uFL3Q7uYAAQigGIRjjMbjOdhs2JAAGt7uVbvdJA9IJwYWTzB07A4APLsdg9MrBAYyIZQcqqGxwACiAA8zpZ9q5JKQ-G0gA)

**Related Issues:**
None that I could find

**Comment by RyanCavanaugh** [comment]
The declaration of `Bar` should be an error, though I'm pretty afraid to change it at this point

**Comment by AnyhowStep** [comment]
When I have to do something like getting the subset of an enum, I just do,

```ts
enum E {
    A = "Hi",
    B = "Bar",
}
const SubEnum : {
    "A" : E.A,
    "B" : E.B
} = {
    A : E.A,
    B : E.B
}
```

With some small effort, you can write a function that creates these sub enums. But it would be nice if what you tried worked correctly, as it's intuitive.

**Comment by trusktr** [comment]
@AnyhowStep But then you can't use `SubEnum` as a type, for example

```ts
function foo(arg: SubEnum) {}
```

won't work the same way as 

```ts
function foo(arg: E) {}
```

Because the first one will expect that you pass in an object containing two properties, while the second one expects that you pass an arg which is one of the things inside the enum.

How do we make a subtype of an enum, so that we can use it like an enum, and it's value are assignable to the superset enum?

**Comment by s-h-a-d-o-w** [comment]
@trusktr I suppose that's a related but still quite different issue than what has been reported here. But I currently have problems with that as well (need to construct subsets of one big string enum that can even overlap at times), so if you've found a solution in the meantime, please let me know.

**Comment by jakebailey** [comment]
Old issue, but this was fixed back in TS 5.0 with all of the new enum behavior updates.
